tests/test.rs

```diff
@@ -956,21 +956,6 @@ fn test_parse_number() {
         ("\t1.0", "invalid number at line 1 column 1"),
         ("1.0\t", "invalid number at line 1 column 4"),
     ]);
-
-    #[cfg(feature = "arbitrary_precision")]
```

Comment:
I removed this test + the extra added function `from_string_unchecked`. In my opinion this test does not makes that much sense at all. Even if we would have parsed the `1e999` correctly into the `Number`, there is not way to get this value out of the object.